File: android/src/main/kotlin/com/usercentrics/sdk/flutter/serializer/UserResponseSerializer.kt
package com.usercentrics.sdk.flutter.serializer

import com.usercentrics.sdk.UsercentricsConsentUserResponse

internal fun UsercentricsConsentUserResponse.serialize(): Any {
    return mapOf(
        "controllerId" to controllerId,
        "userInteraction" to userInteraction.name,
        "consents" to consents.map { it.serialize() }
    )
}

File: ios/usercentrics_sdk/Sources/usercentrics_sdk/Serializer/UserResponseSerializer.swift
import UsercentricsUI
import Flutter

extension UsercentricsConsentUserResponse {
    func serialize() -> Any {
        return [
            "controllerId": self.controllerId,
            "userInteraction": self.userInteraction.serialize(),
            "consents": self.consents.map { $0.serialize() },
        ]
    }
}
